### PR TITLE
Move some cppcheck rules to cppcheck addon instead

### PR DESCRIPTION
**Pull request by pfultz2**
Some other rules can be converted as well but it finds many more cases, so I will enable them in seperate PRs to reduce down the diff.

### CODE REVIEW

File: src/program.cpp [hunk 1]
```diff
@@ -507,10 +507,8 @@ std::vector<argument> generic_eval(const module* mod,
                 }));
         }
         assert(results.find(ins) != results.end());
-        if(not ins->get_shape().any_of_dynamic())
-        {
-            assert(results.at(ins).get_shape() == ins->get_shape());
-        }
+        assert(ins->get_shape().any_of_dynamic() or
```
Comment (TedThemistokleous) [inline src/program.cpp:510]: why drop not here?

**Review by TedThemistokleous — COMMENTED** [review]
Minor question thats it.

File: src/program.cpp (hunk 1 above)
Comment (pfultz2) [reply src/program.cpp:510]: So the `assert` will only assert `results.at(ins).get_shape() == ins->get_shape()` when `ins->get_shape().any_of_dynamic()` is false, to match the previous behavior.

Comment (TedThemistokleous) [reply src/program.cpp:510]: AH you're trying to save a compare. Got it.

**Pull request merged**